sdk/examples/devmode_rust/src/engine.rs

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use std::sync::mpsc::{Receiver, RecvTimeoutError};
+use std::time;
+
+use rand;
+use rand::Rng;
+
+use sawtooth_sdk::consensus::{engine::*, service::Service};
+
+pub struct DevmodeService {
+    pub service: Box<Service>,
+}
+
+impl DevmodeService {
```

Comment:
What is the intention behind this service wrapper?  It looks like it's just ignoring any errors that could occur.

Reply:
The primary motivation was to avoid duplicating the consensus data in `finalize_block` and the vector handling in `check_block` and `get_block`. The others are just to keep things consistent.